receiver/cloudfoundryreceiver/stream.go

```diff
@@ -47,26 +46,36 @@ func newEnvelopeStreamFactory(
 
 func (rgc *EnvelopeStreamFactory) CreateStream(
 	ctx context.Context,
-	shardID string) (loggregator.EnvelopeStream, error) {
+	shardID string,
+	telemetryType telemetryType) (loggregator.EnvelopeStream, error) {
 
-	if strings.TrimSpace(shardID) == "" {
-		return nil, errors.New("shardID cannot be empty")
+	newShardID := shardID
+	selectors := []*loggregator_v2.Selector{}
+	switch telemetryType {
+	case telemetryTypeLogs:
+		newShardID = shardID + "_logs"
```

Comment:
The `shardID` here is something the user customizes in the config. Are we sure we want to modify it no matter what? Especially for existing users sending metrics who will now have a different `shardID` no matter what?

Reply:
This is thetricky part in the implementation. With this current implementation it will be one connection per type of telemetry (logs, metrics, traces) to the loggregator, each connection implements their own filters, so for the loggregator this is like 3 different clients connecting, each one asking for their own telemetry type. 

If we use the same ShardID for 2 types of telemetry (eg, user has a pipeline for metrics and another for logs, both using the same clodfoundryreceiver instance), then each consumer define its filters and we do not know if filters are per-connection. so, What happens if there are 2 connections to loggregator, each with their own filters (logsConsumer defines a filter for logs,  metricsConsumer defines a filter for metrics) but both using the same ShardID?  For example: Is the loggregator going to consider that a logrecord envelope was consumed by the metricsConsumer because is using the same ShardID even having a different filter defined when the connection was created?

In order to be on the safe side, we decided to go with this implementation where each telemetry function uses a different ShardID, because we think is the simple one. The alternative would be using a subroutine which gets all envelope types and routes then with channels to the proper consumer, right?. So with this thought in a balance, we were not sure if such architecture is considered a good practice in the project or because is adding extra complexity can be avoided.

What do you think?

Reply:
Thanks for clarifying, I think I understand your concern here. Since this doesn't actually change existing user functionality we can drop it. 👍